lib/galaxy/tool_util/deps/container_resolvers/mulled.py

```diff
@@ -633,7 +633,15 @@ def pull(self, container):
             cmds = container.build_mulled_singularity_pull_command(
                 cache_directory=self.cache_directory.path, namespace=self.namespace
             )
-            shell(cmds=cmds)
+            # when Galaxy runs in Docker with uswgi, HOME remains set to /root, this is a workaround for that case
```

Comment:
In the Docker-ized Galaxy there is no supervisor involved, so it can't be fixed there  (uwsgi is started by this [script](https://github.com/bgruening/docker-galaxy-stable/blob/master/compose/galaxy-server/files/start.sh#L133) in the docker-compose way of doing things).

I think @nsoranzo's proposed "fake home" fix might work, so I'm going to look into that.

Reply:
That still applies then, you'll want to set the home for other things as well.
Also note that we're not using uwsgi anymore.

Reply:
I presume at some point the docker-galaxy-stable will be updated to not use uwsgi, but that point is likely to be some time in the future. Right now the current release uses Galaxy 20.09 - the fork I am using is using Galaxy 21.01. 

If a workaround is not accepted in the main Galaxy codebase, one option is to patch Galaxy as part of the [Dockerfile](https://github.com/bgruening/docker-galaxy-stable/blob/master/compose/galaxy-server/Dockerfile) in use there, but that's a pretty fragile approach.

Since there is no supervisor config to fix in the docker-compose version of Galaxy I'm using, what do you suggest?

Reply:
Something like `HOME=/home/galaxy ...` in https://github.com/bgruening/docker-galaxy-stable/blob/master/compose/galaxy-server/files/start.sh#L133